### PR TITLE
Add filters for loans and lenders

### LINKED ISSUE #6: Add filtering to get loans endpoint
We need to add filters to the `GET /loans` endpoint so that our frontend will be able to filter results from the endpoint - these filters will be added in another frontend ticket. Exact filtering options are tbd, but can be decided in this ticket - the backend can also support filtering by more than the frontend actually needs.

Resources:
- https://www.django-rest-framework.org/api-guide/filtering/
- https://sunscrapers.com/blog/the-ultimate-tutorial-for-django-rest-framework-filtering-part-5/

Acceptance Criteria:
- Filtering functionality added to `GET /loans` endpoint
  - Supported filters should be decided here as well

### PR DESCRIPTION
**Pull request by skyripley**
## Background
- This PR addresses issue #6, but also adds filters to the `GET /lender/` endpoint. This will allow us to have server-driven filters on our table view in the future. 

## What is changing?
- Adds filters to `GET /loan/` and `GET /lender/` endpoints in the `filters.py` file
  - These filters use [django-filter](https://django-filter.readthedocs.io/en/master/guide/usage.html)
  - The filters expose exact value matching for everything under `Meta` `fields` and also expose custom filters so that we can filter by gte/lte comparisons - these are the more verbose filters in `filters.py`
- Adds api tests and makes some modifications for them to actually run

## How is this tested?
- Api tests
- Tested locally as well

## Notes
- The filters are a bit more verbose than I'd like, but they serve our purpose well
- I opted for api tests rather than unit tests for this functionality because we're using a lot of django specific libraries that need to all work together so to me it didn't make much sense to write unit tests for the classes in `filters.py` as these unit tests wouldn't truly test the functionality

### CODE REVIEW

**Review by Skyblueballykid — APPROVED** [review]
This looks good to me. I will approve and wait for Trang to review before merging.

**Review by ntt2k — APPROVED** [review]
LGTM

**Pull request merged**
